src/Boundary/AdminMapBoundary.java

```diff
@@ -73,6 +76,11 @@ public MapNode newNode(DragIconType type, Point2D loc)
             }
             case Kiosk:
             {
+                Kiosk newKiosk = new Kiosk();
+                Hospital.kiosks.add(newKiosk);
+                n = newKiosk;
```

Comment:
Glad you figured out why its setup like this. (has to do with how a lambda can reference things)